src/Config/ConfigOverrides.php

```diff
@@ -70,10 +120,109 @@ public function loadOverrides($names) {
       }
       $overrides['config_ignore.settings']['ignored_config_entities'] = $existing_ignored;
     }
+
+    // Theme settings override.
+    $this->setLockupOverrides($names, $overrides);
+
+    // GTM overrides.
     $this->setOverridesGoogleTag($names, $overrides);
     return $overrides;
   }
 
+  /**
+   * Disable google tag manager entities when not on prod environment.
+   *
+   * @param array $names
+   *   Config names.
+   * @param array $overrides
+   *   Keyed array of config overrides.
+   */
+  protected function setLockupOverrides(array $names, array &$overrides) {
+
+    // Avoid circular loops.
+    if (!$this->configFactory || in_array('system.theme', $names)) {
+      return;
+    }
+
+    // Validate we are working with the info we need.
+    $theme_info = $this->configFactory->get('system.theme');
```

Comment:
I think the normal approach is to use the `theme_handler` service with the `getDefault()` method.